src/Raven.Server/Documents/Revisions/RevisionsStorage.cs

```diff
@@ -1495,14 +1496,30 @@ private async Task WriteRevertedRevisions(List<Document> list, OperationCancelTo
             list.Clear();
         }
 
-        private bool PrepareRevertedRevisions(DocumentsOperationContext writeCtx, Parameters parameters, RevertResult result, List<Document> list, OperationCancelToken token)
+        private bool PrepareRevertedRevisions(DocumentsOperationContext writeCtx, Parameters parameters, RevertResult result, List<Document> list, OperationCancelToken token, string collection = null)
         {
             using (_database.DocumentsStorage.ContextPool.AllocateOperationContext(out DocumentsOperationContext readCtx))
             using (readCtx.OpenReadTransaction())
             {
-                var revisions = new Table(RevisionsSchema, readCtx.Transaction.InnerTransaction);
-                foreach (var tvr in revisions.SeekBackwardFrom(RevisionsSchema.FixedSizeIndexes[AllRevisionsEtagsSlice],
-                    parameters.LastScannedEtag))
+                IEnumerable<Table.TableValueHolder> tvrs = null;
+                if (collection != null)
+                {
+                    var collectionName = _documentsStorage.GetCollection(collection, throwIfDoesNotExist: false);
+                    if (collectionName == null)
```

Comment:
*Can* this happen? 
What happens here?

If I want to revert two collections, one of them does not exists. Do we fail?
_Why_ fail instead of ignore / log this?

Reply:
If one of the collections doesn't exist we fail.
We can log this instead.

Reply:
Better to log / return the message to the user.

Reply:
Ok